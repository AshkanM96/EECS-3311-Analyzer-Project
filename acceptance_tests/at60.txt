divide
integer_constant(10)
integer_constant(-10)
type_check
evaluate
reset
divide
integer_constant(10)
integer_constant(-3)
type_check
evaluate
