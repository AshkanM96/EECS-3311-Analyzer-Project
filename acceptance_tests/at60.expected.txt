  Expression currently specified: ?
  Report: Expression is initialized.
->divide
  Expression currently specified: (? / nil)
  Report: OK.
->integer_constant(10)
  Expression currently specified: (10 / ?)
  Report: OK.
->integer_constant(-10)
  Expression currently specified: (10 / -10)
  Report: OK.
->type_check
  Expression currently specified: (10 / -10)
  Report: (10 / -10) is type-correct.
->evaluate
  Expression currently specified: (10 / -10)
  Report: -1
->reset
  Expression currently specified: ?
  Report: OK.
->divide
  Expression currently specified: (? / nil)
  Report: OK.
->integer_constant(10)
  Expression currently specified: (10 / ?)
  Report: OK.
->integer_constant(-3)
  Expression currently specified: (10 / -3)
  Report: OK.
->type_check
  Expression currently specified: (10 / -3)
  Report: (10 / -3) is type-correct.
->evaluate
  Expression currently specified: (10 / -3)
  Report: -3
